src/components/ListItem/ListItem.vue

```diff
@@ -0,0 +1,457 @@
+			type: String,
+			required: true,
+		},
+
+		/**
+		* Pass in `true` if you want the matching behaviour to
+		* be non-inclusive: https://router.vuejs.org/api/#exact
+		*/
+		exact: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		* The route for the router link.
+		*/
+		to: {
+			type: [String, Object],
+			default: '',
+		},
+
+		/**
+		 * Id for the <a> element
+		 */
+		anchorId: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * Boldens title and subtitle
+		 */
+		bold: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Toggle the active state of the component.
+		 */
+		active: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Aria label for the wrapper element.
+		 */
+		linkAriaLabel: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * Aria label for the actions toggle.
+		 */
+		actionsAriaLabel: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * If different from from 0 this component will display the
+		 * AppNavigationCounter component
+		 */
+		 counterNumber: {
+			 type: Number,
+			 default: 0,
+		 },
+
+		/**
+		 * Toggles the highlighted state of the counter
+		 */
+		 counterHighlighted: {
+			 type: Boolean,
+			 default: false,
+		 },
+
+	},
+
+	data() {
+		return {
+			hovered: false,
+			focused: false,
+			displayActions: false,
+			menuOpen: false,
+		}
+	},
+
+	computed: {
+
+		hasDetails() {
+			return this.details !== ''
+		},
+
+		hasActions() {
+			return (!!this.$slots.actions)
+		},
+
+		// This is used to decide which outer element type to use
+		// li or router-link
+		navElement() {
+			if (this.to !== '') {
+				return {
+					is: 'router-link',
+					tag: 'li',
+					to: this.to,
+					exact: this.exact,
+				}
+			}
+			return {
+				is: 'li',
+			}
+		},
+
+		// Check if the subtitle slot is populated
+		hasSubtitle() {
+			return !!this.$slots.subtitle
+		},
+	},
+
+	watch: {
+
+		menuOpen(newValue) {
+			console.debug('menuopen', newValue)
+			// A click outside both the menu and the root element hides the actions again
+			if (!newValue && !this.hovered) {
+				this.displayActions = false
+			}
+		},
+	},
+
+	mounted() {
+		console.debug('this.route', this.route, 'this.to', this.to)
+	},
+
+	methods: {
+
+		// forward click event
+		onClick(event) {
+			this.$emit('click', event)
+		},
+
+		handleHover() {
+			this.showActions()
+		},
+
+		handleFocus() {
+			this.focused = true
+			this.showActions()
+		},
+
+		handleBlur(e) {
+			this.focused = false
+		},
+
+		showActions() {
+			if (this.hasActions) {
+				this.displayActions = true
+			}
+		},
+
+		handleMouseleave() {
+			if (!this.menuOpen) {
+				this.displayActions = false
+			}
+		},
+
+		handleTab(e) {
+			if (this.focused && this.hasActions) {
+				console.debug('preventdefault')
+				e.preventDefault()
+				this.$refs.actions.$refs.menuButton.focus()
+				this.focused = false
+			} else {
+				this.displayActions = false
+				this.$refs.actions.$refs.menuButton.blur()
+			}
+		},
```

Comment:
Please add jsDocs to your methods, I have no idea why we're handling the tab here?

You're trying to only show the actions if there was a focus on the item before? Like requiring a double tab? If so I would prefer you to watch the focus event and not the tab. You will cover all the mouse/keyboard use-cases in one go :wink:

Reply:
So the  problem I encountered is I have to display the actions even after the root element is blurred. What I need is: 

* User tabs once
-> the root element is highlighted and the actions are displayed
* User tabs again
-> the main element is blurred and the actions are focused

If I'd only handle the focus event and toggle the actions on and off, the actions would disappear upon pressing tab and would  not be focusable.